osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,156 @@
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Ruleset.BindValueChanged(onRulesetChanged, true);
+        }
+
+        private void onRulesetChanged(ValueChangedEvent<RulesetInfo> ruleset)
+        {
+            SelectedMods.Clear();
+            modsContainer.Clear();
+
+            if (ruleset.NewValue == null)
+                return;
+
+            modsContainer.Add(new ModButton(new NoMod()));
+            modsContainer.AddRange(ruleset.NewValue.CreateInstance().GetAllMods().Where(m => m.Ranked).Select(m => new ModButton(m)));
+
+            modsContainer.ForEach(button => button.OnSelectionChanged = selectionChanged);
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            if (selected)
+                SelectedMods.Add(mod);
+            else
+                SelectedMods.Remove(mod);
+
+            if (!SelectedMods.Any() && !IsHovered)
+                highlightAll();
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            if (!SelectedMods.Any())
+                modsContainer.Children.Where(button => !button.IsHovered).ForEach(button => button.Highlighted.Value = false);
+
+            return base.OnHover(e);
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            base.OnHoverLost(e);
+
+            if (!SelectedMods.Any())
+                highlightAll();
+        }
+
+        public void DeselectAll() => modsContainer.ForEach(mod => mod.Selected.Value = false);
+
+        private void highlightAll() => modsContainer.ForEach(mod => mod.Highlighted.Value = true);
+
+        private class ModButton : ModIcon
+        {
+            private const float mod_scale = 0.4f;
+            private const int duration = 200;
+
+            public readonly BindableBool Selected = new BindableBool();
+            public Action<Mod, bool> OnSelectionChanged;
+
+            public ModButton(Mod mod)
+                : base(mod)
+            {
+                Scale = new Vector2(mod_scale);
+                Highlighted.Value = true;
+                Add(new HoverClickSounds());
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                Selected.BindValueChanged(selected =>
+                {
+                    updateState();
+                    OnSelectionChanged?.Invoke(Mod, selected.NewValue);
+                });
+            }
+
+            protected override bool OnClick(ClickEvent e)
+            {
+                Selected.Value = !Selected.Value;
+                return base.OnClick(e);
+            }
+
+            protected override bool OnHover(HoverEvent e)
+            {
+                updateState();
+                return base.OnHover(e);
```

Comment:
This is sort of correct, as this delegates down to `Drawable` which returns `false`, and it's correct since you are handling `OnHover()` in the parent too to switch all other buttons' highlights. I'd prefer this to be more explicit, however:
```suggestion
                return false;
```